docs/source/api.rst

```diff
@@ -39,6 +39,25 @@ These methods are available directly in the ``ibis`` module namespace.
    trailing_range_window
    random
 
+.. _api.basebackend:
```

Comment:
why does this need to be listed as its internal? (yes its public facing for library extenders but not for users).

Reply:
Backends will be forced to implement them, and the subclass is public. My idea is to document the base class, instead of repeating the documentation for each backend, that will be the same. Also, if we do it this way, backends don't need to write the docs of the methods, which should ve all the same.